src/libstd/ffi/c_str.rs

```diff
@@ -844,6 +950,21 @@ impl CStr {
     ///
     /// [`Cow`]: ../borrow/enum.Cow.html
     /// [`str`]: ../primitive.str.html
+    ///
+    /// # Examples
+    ///
+    /// Calling `to_string_lossy` on a `CStr` with valid unicode:
+    ///
+    /// ```
+    /// use std::borrow::Cow;
+    /// use std::ffi::CStr;
+    ///
+    /// let c_str = CStr::from_bytes_with_nul(b"foo\0").unwrap();
+    /// assert_eq!(c_str.to_string_lossy(), Cow::Borrowed("foo"));
+    /// ```
+    ///
+    /// Had `c_str` contained invalid unicode, the `to_string_lossy` call might
+    /// have returned `Cow::Borrowed("fo�")`.
```

Comment:
It might be worth borrowing from the example in `String::from_utf8_lossy`, which uses the input string `b"Hello \xF0\x90\x80World"` to demonstrate the replacement.